sdk/storage/Azure.Storage.Files.DataLake/src/ErrorExtensions.cs

```diff
@@ -27,12 +28,11 @@ internal static Exception CreateException(this string jsonMessage, ClientDiagnos
             }
             else
             {
-                Dictionary<string, Dictionary<string, string>> errorDictionary
-                    = JsonSerializer.Deserialize<Dictionary<string, Dictionary<string, string>>>(jsonMessage);
+                DataLakeErrorWrapper error = JsonSerializer.Deserialize<DataLakeErrorWrapper>(jsonMessage);
                 return new RequestFailedException(
                     status: response.Status,
-                    errorCode: errorDictionary[Constants.DataLake.ErrorKey][Constants.DataLake.ErrorCodeKey],
-                    message: errorDictionary[Constants.DataLake.ErrorKey][Constants.DataLake.ErrorMessageKey],
+                    errorCode: error.Error.Code,
+                    message: $"{error.Error.Message}\n{error.Error.Detail?.ToErrorDetailString()}",
```

Comment:
I think these belong in the exception's data rather than the message.   It'll still get written out when you call ToString.  See https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs#L73 for more.

Reply:
```
Azure.RequestFailedException: The value for one of the HTTP headers is not in the correct format.
RequestId:4c191595-901f-003b-4877-0c0804000000
Time:2020-04-07T00:56:55.2541980Z ---> System.Exception: Exception of type 'System.Exception' was thrown.
   --- End of inner exception stack trace ---
...
```
I'm not seeing the exceptions data when I'm calling ToString(), after adding the error details to the exception.

Reply:
Sorry, to be more clear, we should use `ClientDiagnostics` to create the exception if possible.

If you do that and pass in the dictionary of additional error details, it'll create both a message in the correct format and add the dictionary to the exception's Data for you.  Then customers get the standard helpful ToString but could programatically access the additional details.

Reply:
Fixed.